### PR TITLE
Migrate Package#build_reason to bootstrap

### PR DESCRIPTION
**Pull request by ChrisBr**
selection_020

selection_018

Getting the table locally can be tricky because you need a package with a dependency on another package, trigger the dependency and then you would get a table. However, you can just mock this by adding this:
```
  def packagechange
    [
      { 'key' => 'package1', 'change' => 'changed' },
      { 'key' => 'package2', 'change' => 'changed' },
      { 'key' => 'package3', 'change' => 'changed' },
      { 'key' => 'package1', 'change' => 'changed' }
    ]
  end
```

to ``src/api/app/models/package_build_reason.rb``.

### CODE REVIEW

File: src/api/app/views/webui2/webui/packages/build_reason/index.html.haml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+.card
+  = render partial: 'webui2/webui/package/tabs', locals: { project: @project, package: @package }
+  .card-body
+    %h3
+      = @pagetitle
+    %h6.subtitle
+      Repository / Architecture: #{@repository.name} / #{@architecture}}
+    %p
+      = link_to '<= Go back to binaries',
+      package_binaries_path(controller: '/webui/package', project: @project, package: @package_name, repository: @repository.name)
```
Comment (DavidKang) [inline src/api/app/views/webui2/webui/packages/build_reason/index.html.haml]: I think the `controller: 'webui/package'` is not needed, right? :thinking: :

Comment (ChrisBr) [reply src/api/app/views/webui2/webui/packages/build_reason/index.html.haml]: Right, removed!

File: src/api/app/views/webui2/webui/packages/build_reason/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+.card
+  = render partial: 'webui2/webui/package/tabs', locals: { project: @project, package: @package }
+  .card-body
+    %h3
+      = @pagetitle
+    %h6.subtitle
+      Repository / Architecture: #{@repository.name} / #{@architecture}}
+    %p
+      = link_to '<= Go back to binaries',
```
Comment (hennevogel) [inline src/api/app/views/webui2/webui/packages/build_reason/index.html.haml]: Please do not reinvent breadcrumbs :-)

Comment (ChrisBr) [reply src/api/app/views/webui2/webui/packages/build_reason/index.html.haml]: Added breadcrumb now instead

**Comment by ChrisBr** [comment]
selection_019

**Comment by mdeniz** [comment]
[nitpick]
Can we do something with that horrible `<= Go back to binaries` link? What about a button with an Icon or other solution?

**Comment by ChrisBr** [comment]
> Can we do something with that horrible <= Go back to binaries link? What about a button with an Icon or other solution?

Already dropped, it is now in the breadcrumbs :+1:

**Comment by mdeniz** [comment]
Update the screenshot then :+1:

**Comment by bgeuken** [comment]
@ChrisBr commented a day ago
> selection_019

IMO the breadcrumb is now overloaded with information. Also when users see this breadcrumb (with the 'Build reason (Tumbleweed / i586)' part) they are already on the build reason page for this particular repo and arch.

**Comment by ChrisBr** [comment]
> IMO the breadcrumb is now overloaded with information. Also when users see this breadcrumb (with the 'Build reason (Tumbleweed / i586)' part) they are already on the build reason page for this particular repo and arch.

True. I'm fine with dropping it, was just like this before...

**Comment by ChrisBr** [comment]
> 'Build reason (Tumbleweed / i586)' part) they are already on the build reason page for this particular repo and arch.

However this is now always the case e.g. look at ``Overview`` we also display this when the user is on the overview page? Maybe we should drop it everywhere to be consistent? Idk ...

Dropped for now ...

**Pull request merged**